sycl/source/detail/scheduler/commands.cpp

```diff
@@ -161,6 +227,212 @@ Command::Command(CommandType Type, QueueImplPtr Queue)
   MEvent.reset(new detail::event_impl(MQueue));
   MEvent->setCommand(this);
   MEvent->setContextImpl(detail::getSyclObjImpl(MQueue->get_context()));
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Obtain the stream ID so all commands can emit traces to that stream
+  if (xptiTraceEnabled()) {
+    MStreamID = xptiRegisterStream(SYCL_STREAM_NAME);
+  }
+#endif
+}
+
+void Command::emitInstrumentationDataProxy() {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  emitInstrumentationData();
+#endif
+}
+
+// Method still takes in void * for the address as adding a template function to
+// the command group object maybe undesirable
+void Command::emitEdgeEventForCommandDependence(Command *Cmd, void *ObjAddr,
+                                                const string_class &Prefix,
+                                                bool IsCommand) {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // If all the information we need for creating an edge event is available,
+  // then go ahead with creating it; if not, bail early!
+  if (xptiTraceEnabled() && MTraceEvent && Cmd && Cmd->MTraceEvent) {
```

Comment:
@romanovvlad  For single check, this may be okay. However, for compound checks, you will end up with a problem if you leave as suggested above. All && should be changed to || and needs a lot of testing/validation for such a simple change.

Reply:
Not sure I understand what you mean. Adding "!" should be enough. But I'm OK if you don't want to do this change.

Reply:
I have made this change.